ui/button.mod/button.js

```diff
@@ -1,485 +1,498 @@
-                    }
-                    if (!this.element.firstChild) {
-                        this.element.appendChild(document.createTextNode(""));
-                    }
-                    this._labelNode = this.element.firstChild;
-                    // this.setLabelInitialValue(this._labelNode.data)
-                    // if (this._label === undefined) {
-                    //     this._label = this._labelNode.data;
-                    // }
-                }
+            const longActionEvent = document.createEvent("CustomEvent");
 
-                //this.classList.add("mod-Button");
-                this.element.setAttribute("role", "button");
-                this.element.addEventListener("keyup", this, false);
-            }
+            // FIXME: InitCustomEvent is deprecated
+            longActionEvent.initCustomEvent("longAction", true, true, null);
+            this.dispatchEvent(longActionEvent);
         }
-    },
+    }
 
     /**
-    Draws the label to the DOM.
-    @function
-    @private
-    */
-    _drawLabel: {
-        enumerable: false,
-        value: function (value) {
-            if(typeof value !== "string") {
-                value = this.defaultLabel;
-            }
-            if (this.isInputElement) {
-                this._element.value = value;
-            } else if (this._labelNode) {
-                this._labelNode.data = value;
+     * Called when all interaction is over.
+     * @protected
+     * @param {MutableEvent} mutableEvent - The event object
+     */
+    handlePressCancel(mutableEvent) {
+        this.active = false;
+        this._removeEventListeners();
+    }
+
+    // <---- Life Cycle ---->
+
+    enterDocument(firstDraw) {
+        super.enterDocument?.call(firstDraw);
+
+        if (firstDraw) {
+            this.element.setAttribute("role", "button");
+
+            const lastChild = this.element.lastChild;
+
+            // Ensure that the last child is a text node
+            // Any whitespace (including indentation) in the template will create a #text node
+            // But just in case (compressed version) we still check if the last child is a text node
+            if (!lastChild || lastChild.nodeType !== Node.TEXT_NODE) {
+                // Create a text node if the last child is not a text node
+                this.element.appendChild(document.createTextNode(""));
             }
-        }
-    },
 
+            this._labelNode = this.element.lastChild;
+
+            // Apply Button styles
+            this._applyImagePositionStyles();
+            this._applyOrientationStyles();
+        }
+    }
 
-    draw: {
-        value: function () {
-            this.super();
-            this._drawLabel(this._label);
+    /**
+     * Draws the component.
+     * @override
+     */
+    draw() {
+        super.draw();
+
+        if (this._labelNode) {
+            this._labelNode.data = this.label;
```

Comment:
Is it really worth giving up backward compatibility and flexibility of being able to use Mod's button with an input type="button"? Pros / cons?